src/pytorch_lightning/callbacks/progress/tqdm_progress.py

```diff
@@ -375,9 +379,7 @@ def convert_inf(x: Optional[Union[int, float]]) -> Optional[Union[int, float]]:
     return x
 
 
-def _update_n(bar: _tqdm, current: int, refresh_rate: int) -> None:
+def _update_n(bar: _tqdm, value: int) -> None:
     if not bar.disable:
-        total = bar.total
-        leftover = current % refresh_rate
-        advance = leftover if (current == total and leftover != 0) else refresh_rate
-        bar.update(advance)
+        bar.n = value
```

Comment:
do you think this is enough to make sure the time estimates are correctly updated?

last time I checked `update` does more than just setting the progress bar counter.
https://github.com/tqdm/tqdm/blob/4f208e72552c4d916aa4fe6a955349ee8b2ed353/tqdm/std.py#L1212-L1277

Reply:
My understanding is that update() was never the problem, even in https://github.com/Lightning-AI/lightning/pull/12889. The problem is in resetting when a new epoch begins. The running estimate that the bar has needs to be reset, otherwise it becomes negative and it would take a few iterations for it to converge to the new value. The only way to achieve this I found is by setting initial = 0. 

[Here](https://github.com/tqdm/tqdm/blob/4f208e72552c4d916aa4fe6a955349ee8b2ed353/tqdm/std.py#L441-L445) is where the computation for the rate happens.
Right below that, the remaining time gets formatted based on the values computed in the rate, hence this is directly affected. 

Note that in your PR in #12889 the value for n is not correct after resuming. The progress bar starts at 0 even though it should start at the step where the checkpoint was saved. This is now fixed by reverting here.

Reply:
I see. At that time, the negative estimates started coming after long training, but that fix introduced this issue where training is resumed.